src/components/App.js

```diff
@@ -1,17 +1,98 @@
+// https://wordsapiv1.p.mashape.com/words/?partOfSpeech=noun
+// https://wordsapiv1.p.mashape.com/words/?partOfSpeech=adjective
+
 import React from 'react';
+import Generate from './Generate';
+import Favourites from './Favourites';
 
+import '../styles/base/base.scss';
 import '../styles/components/app.scss';
 
 class App extends React.Component {
 
   constructor(){
     super();
+    this.state = {
+      type1Result: [], 
+      type2Result: [], 
+      favourites: []
+    } 
+    this.receiveType = this.receiveType.bind(this)
+    this.addFavourites = this.addFavourites.bind(this)
+    this.removeFromFavourites = this.removeFromFavourites.bind(this)
+  }
+
+// want to add this into the fetch content.results[0].definition
+
+  fetchWords(type, source) {
+    let tempArr = [];
+    const wordsUrl = 
+    `https://wordsapiv1.p.mashape.com/words/?partOfSpeech=${type}&random=true`
+    
+    for(let i=0; i < 3; i++) {
+      fetch(wordsUrl, { headers: { 
+        "X-Mashape-Key": "wZCRY1mr7wmsh03nJhT5o7UZZ6uYp16uOdRjsnMQ91Z2XQEGFb", 
+        "X-Mashape-Host": "wordsapiv1.p.mashape.com"
+      } } )
+      .then(response => response.json())
+      .then(content => {
+        tempArr = tempArr.concat([content.word]);
+        if(source === 'type1') {
+          this.setState({type1Result: tempArr});
+        } 
+        if(source === 'type2') {
+          this.setState({type2Result: tempArr});
+        }
+      })
+
+  }
+}
+
+  receiveType(type, source) {
+    this.fetchWords(type, source)
+  }
+
+  addFavourites(corpseitem) {
+    this.setState({
```

Comment:
rather than using a callback here, it would be better to use 

```
const favourites = this.state.favourites.concat(corpseitem);
this.setState({
  favourites: favourites
});
window.localStorage.setItem('favourites', favourites);
```